src/aws-codeartifact/domain.ts

```diff
@@ -0,0 +1,213 @@
+import { ArnFormat, IResource, ITaggableV2, Resource, Stack, TagManager, TagType } from 'aws-cdk-lib';
+import { CfnDomain, CfnDomainProps } from 'aws-cdk-lib/aws-codeartifact';
+import {
+  AddToResourcePolicyResult,
+  Grant,
+  GrantWithResourceOptions,
+  IGrantable,
+  PolicyDocument,
+  PolicyStatement,
+  principalIsOwnedResource,
+} from 'aws-cdk-lib/aws-iam';
+import { IKey, Key } from 'aws-cdk-lib/aws-kms';
+import { Construct } from 'constructs';
+
+/**
+ * Represents an Codeartifact Domain
+ */
+export interface IDomain extends IResource {
+  /**
+   * The ARN of the Domain
+   *
+   * @attribute
+   */
+  readonly domainArn: string;
+
+  /**
+   * The name of the Domain
+   *
+   * @attribute
+   */
+  readonly domainName: string;
+
+  /**
+   * The key used to encrypt the Domain
+   *
+   * @attribute
+   */
+  readonly encryptionKeyArn?: string;
```

Comment:
```suggestion
  /**
   * The key used to encrypt the Domain
   *
   * @attribute
   */
  readonly encryptionKey?: IKey;
```

Reply:
How do you suggest to also comply with the AWS CDK design guidelines? Expose both the `IKey` and attribute? At least that's what I am thinking.

> AWS constructs must expose all resource attributes defined in the underlying
CloudFormation resource as readonly properties of the class
_[awslint:resource-attribute]_.

https://github.com/aws/aws-cdk/blob/b49032b3a6e549783b45492ffc76880fbcd58e68/docs/DESIGN_GUIDELINES.md?plain=1#L684-L686

Reply:
Makes sense, I actually mis-read it for props, but this is the interface. In which case I'd say my suggestion is wrong, and we should not replace the prop with `IKey`. But optionally we _could_ expose the `IKey` prop _in addition_ to the string value.